include/zephyr/dsp/basicmath.h

```diff
@@ -0,0 +1,915 @@
+/* Copyright (c) 2022 Google LLC
```

Comment:
should probably add ARM copyright here since this code is more or less directly based on their CMSIS-DSP headers.

Reply:
Done